Signal/src/call/PeerConnectionClient.swift

```diff
@@ -70,12 +72,24 @@ class PeerConnectionClient: NSObject, RTCPeerConnectionDelegate, RTCDataChannelD
              dataChannelSignaling = "signaling"
     }
 
+    // A state in this class should only be accessed on this queue in order to
+    // serialize access.
+    //
+    // This queue is also used to perform expensive calls to the WebRTC API.
+    private static let signalingQueue = DispatchQueue(label: "CallServiceSignalingQueue")
+
     // Delegate is notified of key events in the call lifecycle.
-    public weak var delegate: PeerConnectionClientDelegate!
+    private weak var delegate: PeerConnectionClientDelegate!
+
+    func setDelegate(delegate: PeerConnectionClientDelegate?) {
+        PeerConnectionClient.signalingQueue.async {
+            self.delegate = delegate
+        }
+    }
 
     // Connection
 
-    internal var peerConnection: RTCPeerConnection!
+    private var peerConnection: RTCPeerConnection!
```

Comment:
Ah, and I've broken the tests.  Fixed.  I've added test-specific accessors/utility methods to avoid having production code use these pathways.